_posts/2018-07-10-react-native-wdio.md

```diff
@@ -0,0 +1,321 @@
+
+**Step 3:** Upload your `.app` or `.apk` file to SauceStorage or any other accessible endpoint and then update your app path in capabilities to point to the remote storage location (as seen below). You can follow the documentation here; [Uploading Mobile Applications to Sauce Storage for Testing](https://wiki.saucelabs.com/display/DOCS/Uploading+Mobile+Applications+to+Sauce+Storage+for+Testing) for more information around uploading your app to Sauce Labs.
+
+```diff
+- app: <app path>
++ app: <sauce-storage:myapp.zip or app link>
+```
+
+**Run the test:**
+Let's run the test using Sauce Labs with the updated configuration.
+
+```console
+wdio wdio.conf.js
+```
+
+Check the [Sauce Labs Dashboard](https://saucelabs.com/beta/dashboard/tests) to make sure the test ran successfully. The WebDriverIO Sauce service automatically sets test labels and results. Now that you have run your test on could based emulators, this opens up the ability to scale out the automation of your functional tests. Sometimes, an emulator doesn't quite meet all the requirements necessary to ensure confidence in app quality due to real sensors, older devices, and other aspects like memory and CPU of a real device.
+
+## Running tests on real devices using a cloud-based service
+
+Sauce Labs also provides a real device testing solution through the TestObject platform. With some minor changes to the above `wdio.conf`, we can run UI tests on real devices.
+
+First, create a project in the [TestObject dashboard](https://app.testobject.com/) and upload your `.ipa` or `.apk` file.
+
+There is no official WebDriverIO TestObject service and at the time of writing this blog, Sauce Labs and TestObject do not share the same API calls.
+
+**Update `wdio.conf` file:**
+
+```diff
+- services: ['sauce'],
++ protocol: 'https',
++ host: 'us1.appium.testobject.com',
+- port: '80',
++ port: '443',
++ path: '/api/appium/wd/hub',
+capabilities: {
+  ...
+-  app: 'sauce-storage:myapp.zip'
++  testobject_api_key: <TESTOBJECT_ACCESS_KEY>,
++  testobject_app_id: <APP_ID>,
+}
+```
+
+Refer to [Appium Capabilities for Real Device Testing](https://wiki.saucelabs.com/display/DOCS/Appium+Capabilities+for+Real+Device+Testing) to update your capabilities if needed.
+
+**Run the test:**
+Let's run the test with the updated configuration.
+
+```console
+wdio wdio.conf
+```
+
+Check the [TestObject Dashboard](https://app.testobject.com/) to make sure the test has run. WebDriverIO does not update test labels or results in TestObject Dashboard. TestObject does not aggregate the test results when running through WebDriverIO, therefore, we have to write some extra code so that your test results are in sync. Here are some references that can help with that: <https://github.com/pizzasaurusrex/TestObject> or <https://gist.github.com/rajapanidepu/0e8c0f89671a8a563a7463f8c1ff0413>
+
```

Comment:
Run on sentence
> TestObject does not aggregate the test results when running through WebDriverIO, therefore, we have to write some extra code so that your test results are in sync.

Should be (I also edited this for clarity): 
TestObject does not aggregate the test results when running through WebDriverIO. Therefore, write some extra code to keep test results in sync.